connectors/standalone/src/main/scala/io/delta/standalone/internal/OptimisticTransactionImpl.scala

```diff
@@ -408,8 +409,12 @@ private[internal] class OptimisticTransactionImpl(
    * Looks at actions that have happened since the txn started and checks for logical
    * conflicts with the read/writes. If no conflicts are found return the commit version to attempt
    * next.
+   *
+   * @throws DeltaConcurrentModificationException if the next commit version cannot be resolved
+   *                                              due to the concurrent modifications in the
+   *                                              commit log.
    */
-  private def checkForConflicts(
+  private def getNextCommitVersionOrFailIfConflictsDetected(
```

Comment:
rant: Just this name itself shows how much it "begs" for some refactoring so it does not do multiple things at once, huh?! 😉

Reply:
Agree, a function doing 2 things is harder to read than a single responsibility one. However, here the version increment is conditioned by the lack of conflicts, so it acts as an inherent validation part of the process. It did surprise me in the beginning but now it's not. 

Eventually the method could be simply called `getNextCommitVersion` but the detail that it may fail is then hidden... Unless we had _checked exceptions_ in Scala :stuck_out_tongue_winking_eye: